### PR TITLE
Flax implementation of DPT

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

I tried to implement DPT (Dense Prediction with Transformers) in Flax during my free time! 🚀 
By the way it is the first Segmentation and Depth Estimation model implemented in Flax!

Nits/TODOs:
- [x] Figure out how to properly call `BatchNorm` and `Dropout` inside a `Sequential` 
- [ ] Test equivalency tests 
- [ ] Write documentation - For now they're just copy/pasted
Quetions:
- Why the loss is not implemented in `modeling_dpt.py` ? I can probably help on that since I have already implemented the loss for a university project: https://github.com/antocad/FocusOnDepth/blob/master/FOD/Loss.py 
 
cc @NielsRogge @sanchit-gandhi @patil-suraj

### CODE REVIEW

File: src/transformers/models/dpt/modeling_flax_dpt.py [hunk 1]
```diff
@@ -0,0 +1,1121 @@
...
+            tensors for more detail.
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors for
+            more detail.
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+"""
+
+
+class FlaxPatchEmbeddings(nn.Module):
```
Comment (ArthurZucker) [inline src/transformers/models/dpt/modeling_flax_dpt.py:82]: Missing `#copied from FlaxViT` as I think it is copy pasted

**Review by ArthurZucker — COMMENTED** [review]
Gonna have a deeper look later on
